src/lang/modifyAst/addFillet.test.ts

```diff
@@ -313,3 +318,82 @@ const extrude001 = extrude(-5, sketch001)
     expect(edges).toEqual([])
   })
 })
+
+describe('Testing button states', () => {
+  const runButtonStateTest = async (
+    code: string,
+    segmentSnippet: string,
+    expectedState: boolean
+  ) => {
+    // ast
+    const astOrError = parse(code)
+    if (astOrError instanceof Error) {
+      return new Error('AST not found')
+    }
+    const ast = astOrError as Program
+
+    // selectionRanges
+    const range: [number, number] = segmentSnippet
+      ? [
+          code.indexOf(segmentSnippet),
+          code.indexOf(segmentSnippet) + segmentSnippet.length,
+        ]
+      : [ast.end, ast.end] // empty line in the end of the code
+
+    const selectionRanges: Selections = {
+      codeBasedSelections: [
+        {
+          range,
+          type: 'default',
+        },
+      ],
+      otherSelections: [],
+    }
+
+    // state
+    const buttonState = hasValidFilletSelection({
+      ast,
+      selectionRanges,
+      code,
+    })
+
+    expect(buttonState).toEqual(expectedState)
+  }
+  const codeWithBody: string = `
+    const sketch001 = startSketchOn('XY')
+      |> startProfileAt([-20, -5], %)
+      |> line([0, 10], %)
+      |> line([10, 0], %)
+      |> line([0, -10], %)
+      |> lineTo([profileStartX(%), profileStartY(%)], %)
+      |> close(%)
+    const extrude001 = extrude(-10, sketch001)
+  `
+  const codeWithoutBodies: string = `
+    const sketch001 = startSketchOn('XY')
+      |> startProfileAt([-20, -5], %)
+      |> line([0, 10], %)
+      |> line([10, 0], %)
+      |> line([0, -10], %)
+      |> lineTo([profileStartX(%), profileStartY(%)], %)
+      |> close(%)
+  `
+  // body is missing
+  it('should return false when body is missing and nothing is selected', async () => {
+    await runButtonStateTest(codeWithoutBodies, '', false)
+  })
+  it('should return false when body is missing and segment is selected', async () => {
+    await runButtonStateTest(codeWithoutBodies, `line([10, 0], %)`, false)
+  })
+
+  // body exists
+  it('should return true when body exists and nothing is selected', async () => {
+    await runButtonStateTest(codeWithBody, '', true)
+  })
+  it('should return true when body exists and segment is selected', async () => {
+    await runButtonStateTest(codeWithBody, `line([10, 0], %)`, true)
+  })
+  it('hould return false when body exists and not a segment is selected', async () => {
+    await runButtonStateTest(codeWithBody, `close(%)`, false)
+  })
```

Comment:
I think we should merge this anyway, because this represents current behaviour, but we might want to allow fillets on `close` at some point though it's tricky.

So this is legitimate working code
![image](https://github.com/user-attachments/assets/aabf05af-6eb8-4f6d-b74b-6c617057a7eb)


however this is hand tweak code because normal kclSketches will have `lineTo([profileStartX(%), profileStartY(%)], %)` in them, making the `close` a no-op. which is also why I think leave it fro now, but wanted to leave an FYI.

Reply:
wow, thanks, didn't know that

Reply:
@Irev-Dev why do we need `lineTo([profileStartX(%), profileStartY(%)], %)` if we can replace it with `close` and other way round, why do we need `close` if profile anyway closes with `lineTo([profileStartX(%), profileStartY(%)], %)` ? Won't it be more straightforward to just have one universal way to do it?

Reply:
@franknoirot correct me if I'm wrong.

We wanted to have `lineTo([profileStartX(%), profileStartY(%)], %)` because I guess it's more expressive, but it's risky because we're gambling with the engine's resolution being the same as kcl, where as close is explicit and safe to close the profile.